lib/docker2aci.go

```diff
@@ -167,24 +166,39 @@ func SquashLayers(images []acirenderer.Image, aciRegistry acirenderer.ACIRegistr
 	}
 
 	squashedFilename := getSquashedFilename(parsedDockerURL)
-	squashedImagePath := path.Join(outputDir, squashedFilename)
+	squashedImagePath := filepath.Join(outputDir, squashedFilename)
 
-	squashedImageFile, err := os.Create(squashedImagePath)
+	squashedTempFile, err := ioutil.TempFile(outputDir, "docker2aci-squashedFile-")
 	if err != nil {
 		return "", err
 	}
-	defer squashedImageFile.Close()
+	defer func() {
+		errClose := squashedTempFile.Close()
+		if err != nil {
+			// remove temp file on error
+			errRemove := os.Remove(squashedTempFile.Name())
+			if errRemove != nil {
+				err = errRemove
```

Comment:
> I think it's important to not ignore the Close error on successful cases to avoid data corruption.

You're right.


> How about we remove the temp file on error ignoring the Remove errors so we don't mask the "real" one and we handle the close error if there was a successful execution?

Seems Good!